### DISCUSSION TITLE
nginx config for playing videos on firefox/safari

### DISCUSSION THREAD

**Discussion by harshavmb** [photoprism/photoprism#2964] [General]
Hi,

Though it's not directly related to Photoprism, I think many people could be facing this problem. I've been using Chrome & mobile app all these days. 

Having got a new Mac M1, I'm puzzled to see videos not playing on Safari & Firefox browser when I proxy via Nginx. Nevertheless, the same video plays fine on Chrome. Also, videos play fine when I directly access photoprism via the LAN endpoint. 

Having triaged a while, I think it's much to do with `slices` & `range` headers required by these browsers. I did try various configurations, saw few improvements but never matched Chrome. Has someone got the right nginx config for these browsers?

I'm not caching anything here. Just reverse proxying the photoprism server. 

Below is the configuration, I've tried::

    client_max_body_size 15000M;

    listen 443 ssl;
    # Update this line to be your domain
    server_name <my-domain>;

    # Ensure these lines point to your SSL certificate and key
    ssl_certificate /etc/letsencrypt/live/<my-domain>/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/<my-domain>/privkey.pem; # managed by Certbot

    add_header Strict-Transport-Security "max-age=31536000; includeSubdomains";
    # ssl on; # Uncomment if you are using nginx < 1.15.0
    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!RC4";
    ssl_prefer_server_ciphers off;
    ssl_session_cache shared:SSL:10m;

    proxy_buffering on;
    proxy_force_ranges on;
    slice 1m;

    location / {

        proxy_pass http://127.0.0.1:2342;
        proxy_set_header Host $host;
        proxy_set_header Range $slice_range;
        proxy_http_version 1.1;
        #proxy_read_timeout 300;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
    }

**Comment by srett** [comment]
That's what I'm using, and I exclusively use Firefox on all devices:

```
location / {
   allow 127.0.0.1;
   allow ::1;
   allow 192.168.0.0/24;
   deny all;
   proxy_pass http://127.0.0.1:2342;
   proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
   proxy_set_header Host $host;
   proxy_buffering off;
   proxy_http_version 1.1;
   proxy_set_header Upgrade $http_upgrade;
   proxy_set_header Connection "upgrade";
}
```

**Reply by harshavmb** [reply]
Thanks,

I've tried with `proxy_buffering` off but no improvements. For some reason, firefox doesn't fetch the video content like chrome. It's painfully slow. 

I also see `NS_ERROR_PARSED_DATA_CACHED` in the browser console. Even 20 MB videos take ages to play.

**Comment by russeldsouzaFRM** [comment]
I'm running a React JS application on an Nginx server (Ubuntu 22.04) and encountering problems with video playback on iOS device. The videos play correctly on other platforms. Below I've attached screenshot.

![ios-ss](https://github.com/user-attachments/assets/210a4b20-0328-4102-a92f-dcd12316b592)
